lib/HALSim/src/Network.h

```diff
@@ -99,9 +99,9 @@ class Network : public INetwork
     /**
      * Get the IP address.
      *
-     * @return IP address if available, otherwise an empty string.
+     * @return IP address if available, otherwise IPAddr(), which is 0.0.0.0.
      */
-    String getIp() const final;
+    IPAddress getIp() const;
```

Comment:
Shouldn't this be `final` ?

Reply:
What's the benefit ? It might help optimizing if called a million times. But then the client code doesn't seem to care and the valueof a having  an INetwork interface is diminishing.

Is there a CS rule when to apply final ?

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-override

Reply:
The rule is here to mark it with "final" to specify its the only and final overrider.

Reply:
final added